### DISCUSSION TITLE
NextAuth session callback only changing focus

### DISCUSSION THREAD

**Discussion by Naghell** [nextauthjs/next-auth#9342] [Help]
Hi! I apologize in advance for the wording of this text. Please note that everything went through Google Translator.

I am having a problem with my application in Next14. Let's say I'm in an administrative dashboard which allows access only to logged in users and, depending on the information of each user, displays different elements. Everything works fine but the problem is when I refresh, or exit the page and enter again. The session loads halfway but the elements or all the information is not displayed until I change tabs and go back, or simply focus somewhere else. I checked in network and indeed, session is not called until I change the focus. Do you have any idea what it could be? I'm 90% sure it's a NextAuth problem but I tried too much and can't fix it.
I clarify that this only occurs in production or in build. If I try the application in npm run dev this problem is not replicated.

**Comment by rafin-rahman** [comment]
I'm encountering a similar problem and haven't been able to find a solution yet. The main issue is that during the session loading, only the default user information (such as NAME, EMAIL, IMAGE) is loaded. Any additional custom properties that are added do not seem to be recognised immediately. 

The only workaround I've noticed is when I switch tabs or shift the focus from the browser to a different application and then return. If you check the network tab, it becomes evident that the session information is actually loaded only when you switch between tabs or applications.

![image](https://github.com/nextauthjs/next-auth/assets/38799209/827ecf47-b937-40d1-b963-5a6dcea9dae2)

**Reply by ronybarua-ethical18** [reply]
did you find any solution?

**Reply by rafin-rahman** [reply]
Nope, I’ve ended up making my own RBAC system. 
nextauth got poor documentation, the time you spend understanding it is the same as making your own auth system.

**Comment by kuptiman** [comment]
I have the same problem, I think it is related to #596 - as of **v4.22.1** there is an Session _update()_ API see the updated [docs](https://next-auth.js.org/getting-started/client#updating-the-session), or you can try this workaround which worked for me: https://github.com/nextauthjs/next-auth/issues/596#issuecomment-1416436344
